pkg/core/chain/chain.go

```diff
@@ -787,3 +780,19 @@ func mockFirstIntermediateBlock(prevBlockHeader *block.Header) (*block.Block, er
 
 	return blk, nil
 }
+
+func mockDeterministicCoinbase() transactions.Transaction {
+	seed := make([]byte, 32)
+
+	keyPair := key.NewKeyPair(seed)
+	tx := transactions.NewCoinbase(make([]byte, 32), make([]byte, 32), 2)
+	var r ristretto.Scalar
+	r.SetZero()
+	tx.SetTxPubKey(r)
+
+	var reward ristretto.Scalar
+	reward.SetBigInt(big.NewInt(int64(config.GeneratorReward)))
```

Comment:
The reward should be a callback injected in this (and the other reward aware) functions, also to allow for dynamic reward adjusting.

Reply:
Can be also tackled in a separate issue/PR, as long as this is recorded

Reply:
Opened in #238